File: Semester 1/Class03_Computer_Contest_Lv_2_Notes_and_Homework_Sep_25-26/lvl2_hw_sol/lvl2_railway_sort.cpp
#include <bits/stdc++.h>
using namespace std;
int n;
int main(){
    for(int t=1; t<=10; t++) {
        cin >> n; vector<int> car;
        for(int i=0, x; i<n; i++) {
            cin >> x; car.push_back(x);
        }
        int cost = 0;
        for(int i=n-1; i>=1; i--) {
            auto cur = find(car.begin(), car.end(), i), nxt = find(car.begin(), car.end(), i+1);
            if(cur > nxt) {
                cost += cur - car.begin();
                car.erase(cur);
                car.insert(car.begin(), i);
            }
        }
        cout << cost << endl;
    }
}

File: Semester 1/Class03_Computer_Contest_Lv_2_Notes_and_Homework_Sep_25-26/lvl2_hw_sol/lvl2_railway_sort.py
for t in range(10):
    n = int(input())
    car = list(map(int, input().split()))
    cost = 0
    for i in range(n-1, 0, -1):
        cur, nxt= car.index(i), car.index(i+1)
        if cur > nxt:
            cost += cur
            car.pop(cur)
            car.insert(0, i)
    print(cost)